flytekit/clients/raw.py

```diff
@@ -1,17 +1,46 @@
 from __future__ import absolute_import
+
 from grpc import insecure_channel as _insecure_channel, secure_channel as _secure_channel, RpcError as _RpcError, \
     StatusCode as _GrpcStatusCode, ssl_channel_credentials as _ssl_channel_credentials
 from flyteidl.service import admin_pb2_grpc as _admin_service
 from flytekit.common.exceptions import user as _user_exceptions
 import six as _six
+from flytekit.configuration import creds as _creds_config, platform as _platform_config
+
+from flytekit.clis.auth import credentials as _credentials_access
+from flytekit.clients.helpers import (
+    get_global_access_token as _get_global_access_token, set_global_access_token as _set_global_access_token
+)
+
+
+def _try_three_times(fn):
+    def handler(*args, **kwargs):
+        attempt = 0
+        while True:
+            try:
+                attempt += 1
+                return fn(*args, **kwargs)
+            except Exception as e:
+                if attempt >= 3:
+                    raise e
+                else:
+                    print('retrying')
+    return handler
 
 
 def _handle_rpc_error(fn):
+    @_try_three_times
```

Comment:
hey @matthewphsmith we introduced this to make sure we don't get stuck in a loop if authentication is persistently failing...but i'm not sure that this is the prettiest code 🙃 let us know if a better way stands out

Reply:
the code's been refactored but i think the original comment still stands